### PR TITLE
Improve nullity checking

### PR DESCRIPTION
**Pull request by blairconrad**
Hey. I'd had concerns about argument null checks, specifically with FakeItEasy, so I added a few more tests, some of which broke. For FakeItEasy. See what you think.

I made a couple of bonus commits first. I can move them to a different PR(s), if you prefer.

### CODE REVIEW

File: Source/Machine.Fakes.Adapters.Specs/InlineConstraintSpecs.generated.cs [hunk 1]
```diff
@@ -35,7 +35,46 @@ public class When_matching_any_parameter_value_and_passing_null : WithCurrentEng
 
         Because of = () => _configuredBehaviorWasTriggered = _view.TryLogin(null, null);
 
-        It should_have_trigged_the_configured_behavior = () => _configuredBehaviorWasTriggered.ShouldBeTrue();
+        It should_have_triggered_the_configured_behavior = () => _configuredBehaviorWasTriggered.ShouldBeTrue();
```
Comment (blairconrad) [inline Source/Machine.Fakes.Adapters.Specs/InlineConstraintSpecs.generated.cs]: I thought "trigged" was a typo.

File: Source/Machine.Fakes.Adapters.Specs/InlineConstraintSpecs.tt [hunk 2]
```diff
@@ -6,7 +6,7 @@
 <#@ output extension=".generated.cs" #>
 <# var allEngines = new[] { "RhinoFakeEngine", "NSubstituteEngine", "MoqFakeEngine", "FakeItEasyEngine" };  #>
 <# var allNamespaces = new[] { "Rhinomocks", "NSubstitute", "Moq", "FakeItEasy" };  #>
-<# var supportsNetStandard = new[] { false, false, true, false };  #>
+<# var supportsNetStandard = new[] { false, true, true, true };  #>
```
Comment (blairconrad) [inline Source/Machine.Fakes.Adapters.Specs/InlineConstraintSpecs.tt]: It looked to me like the generated test file was saved, adding NSubstitute and FakeItEasy to those tested under .NET Standard, but this file wasn't updated. Changing these values here mean we _keep_ testing those two frameworks under .NET Standard.

File: Source/Machine.Fakes.FakeItEasy/FakeItEasyExpressionRewriter.cs [hunk 3]
```diff
@@ -63,28 +63,26 @@ static Expression RewriteIsMethod(MethodCallExpression expression)
...
+            {
+                // If we're checking a value type for nullity, it means the type
+                // is Nullable<T>, and we want to find IsNull<T>. (Or IsNotNull<T>.) 
+                argumentType = argumentType.GetGenericArguments().First();
+            }
 
-            var method = GetIsNullMethod(argumentType);
+            var method = typeof(ArgumentConstraintManagerExtensions)
+                .GetMethods()
+                .Where(x => x.Name == node.Member.Name)
```
Comment (blairconrad) [inline Source/Machine.Fakes.FakeItEasy/FakeItEasyExpressionRewriter.cs]: I'm taking advantage of

1. the new `IsNotNull` method, and
2. the fact that the Machine.Specs constraint names are the same as FakeItEasy's.

If it's too cutesy, I can make more explicit.

**Comment by ivanz** [comment]
Thank you!

**Comment by blairconrad** [comment]
A pleasure. Thanks for the merge, @ivanz.

**Pull request merged**